library/src/cache/disk_io.rs

```diff
@@ -0,0 +1,104 @@
+use anyhow::{bail, Context};
+use serde::{de::DeserializeOwned, Serialize};
+use std::{
+    fs::File,
+    io::{BufReader, BufWriter},
+    path::Path,
+};
+
+// https://stackoverflow.com/questions/67087597/is-it-possible-to-use-rusts-log-info-for-tests
+#[cfg(test)]
+use std::println as debug; // Workaround to use println! for logs.
+
+pub fn write<S, P>(serializable: &S, path: &P) -> anyhow::Result<()>
+where
+    S: ?Sized + Serialize,
+    P: AsRef<Path>,
+{
+    debug!("Writing to {:?}", path.as_ref());
+
+    let path_as_ref = path.as_ref();
+    let containing_dir = path_as_ref
+        .parent()
+        .with_context(|| format!("Failed to get parent dir for {:?}", path_as_ref))?;
+
+    // Because File::create can sometimes fail if the full directory path doesn't exist,
+    // we create the directories in its path first.
+    std::fs::create_dir_all(containing_dir).with_context(|| {
+        format!(
```

Comment:
CodeCov says you missed the failure case.  Might be worth testing?

Reply:
Probably can just create a directory or path in the way and drop write permissions on it?

Reply:
This file has 100% coverage according to the https://app.codecov.io/gh/shorebirdtech/updater/pull/93/blob/library/src/cache/disk_io.rs, resolving